contributing/code/conventions.rst

```diff
@@ -79,31 +79,45 @@ must be used instead (where ``XXX`` is the name of the related thing):
 
 .. _contributing-code-conventions-deprecations:
 
-Deprecations
-------------
+Deprecating code
+----------------
 
 From time to time, some classes and/or methods are deprecated in the
 framework; that happens when a feature implementation cannot be changed
 because of backward compatibility issues, but we still want to propose a
 "better" alternative. In that case, the old implementation can simply be
 **deprecated**.
 
+Deprecations must only be introduced on the next minor version of the impacted
+component (or bundle, bridge, contract).
+They can exceptionally be introduced on previous supported versions if they are critical.
+
+A new class (or interface, trait) cannot be introduced as deprecated, or
+contains deprecated methods.
+
 A feature is marked as deprecated by adding a ``@deprecated`` phpdoc to
 relevant classes, methods, properties, ...::
 
     /**
-     * @deprecated since vendor-name/package-name 2.8, to be removed in 3.0. Use XXX instead.
+     * @deprecated since Symfony 2.8.
```

Comment:
Why the rewording here?

cc @greg0ire

Reply:
I added this after talking with @nicolasgrekas, what's the deal?

Reply:
Yes you added the vendor/... thingy which now gets removed, that’s my point

Reply:
I know, I was more asking @fancyweb or @nicolas-grekas to explain the sudden change of heart

Reply:
I think the misunderstanding is because this is the "Contribute to Symfony" doc ... so it's not a generic-purpose doc and we don't need things like "vendor-name/package-name" because "vendor" is always "Symfony" and the "package-name" is not usually included in the Symfony deprecation messages.

Reply:
we're now getting rid of the `to be removed` part as it doesn't bring much value
about vendors, @javiereguiluz is right, but it'd be nice to hint ppl that for other vendors, the package should really be mentionned
maybe as an rst note?

Reply:
Regarding the package name, there is not always a way to guess it from a class name, for instance here:

https://github.com/symfony/var-exporter/blob/8539c2cec7202d244058075351c61aa852ffa344/Internal/Exporter.php#L120


Also, why remove the `Use XXX instead.` part? If there is a replacement, it should absolutely be mentioned IMO.

Reply:
The "use XXX instead" is not removed. This example is just below.

Reply:
@greg0ire this link points to a notice, not to a deprecation :)

Reply:
Oh right my bad, here is another example:

https://github.com/symfony/symfony/blob/38143c604f68038e28cf2fa2fbb61c04fc68b98c/src/Symfony/Component/Serializer/Encoder/CsvEncoder.php#L51

Reply:
bad one indeed :)